libs/accounts/recovery-phone/src/lib/recovery-phone.service.ts

```diff
@@ -109,26 +116,24 @@ export class RecoveryPhoneService {
       throw new RecoveryPhoneRegistrationLimitReached(phoneNumber);
     }
 
+    // Create a code and store it for validation later
     const code = await this.otpCode.generateCode();
-
     await this.recoveryPhoneManager.storeUnconfirmed(
       uid,
       code,
       phoneNumber,
       true
     );
 
-    const formattedSMSbody = getFormattedMessage
-      ? await getFormattedMessage(code)
-      : undefined;
-
-    const smsBody = formattedSMSbody || `${code}`;
-
+    // Pick the message with the best length and send it
+    const formattedMessages = await getFormattedMessages(code);
+    const body = this.getSafeSmsBody(formattedMessages);
     const msg = await this.smsManager.sendSMS({
```

Comment:
Should this be wrapped in a try/catch block to prevent `MessageBodyTooLong` errors (though unlikely) from bubbling up to the user?

Reply:
I'll punt on that for now, maybe ok to fail hard and fast here.